src/traccuracy/_tracking_graph.py

```diff
@@ -145,7 +145,17 @@ def __init__(
             name (str, optional): User specified name that will be included in result
                 outputs associated with this object
         """
-        self.segmentation = segmentation
+
+        # if the segmentation is not integer dtype, cast to uint64. Otherwise, leave it
```

Comment:
A statement to confirm that my understanding of the implications of this change are correct:

This could result in a situation where the type of the `label_id` on the graph is different from the type of the segmentation arrays. However, this should be fine because numpy seems to generally be sensible about comparing between floats and ints.

Reply:
Yes. But, if we implement the casting in the loaders, the loaders can ensure that the label_id is consistent with the segmentation.